base/sg_cluster_cfg.go

```diff
@@ -51,19 +52,23 @@ func NewCfgSG(bucket Bucket) (*CfgSG, error) {
 		bucket:        bucket,
 		loggingCtx:    loggingCtx,
 		subscriptions: make(map[string][]chan<- cbgt.CfgEvent),
+		groupID:       groupID,
 	}
 	return c, nil
 }
 
-func sgCfgBucketKey(cfgKey string) string {
+func (c *CfgSG) sgCfgBucketKey(cfgKey string) string {
+	if c.groupID != nil {
+		return SGCfgPrefix + ":" + *c.groupID + ":" + cfgKey
```

Comment:
Don't we have a potential collision here between the `(groupID, cfgKey)` pairs `(nil, "a:b")` and `("a","b")`?  I don't think have any restrictions on cfgKey that would prevent `a:b` as a value.

Reply:
This is a potential collision. @bbrks and I discussed this and decided the best solution that keeps backwards compatibility is to use a different constant prefix (that's created during runtime using the group ID) if a group ID is used. 

For example: (groupID, cfgKey) pairs ("a","b") would be `_sync:a:cfg:b` for CfgSG. Pairs (nil, "a:b") would be `_sync:cfga:b` which is the same as it was in legacy, while avoiding a collision. 

This will also be used in other places where namespacing is being added such as the sync function and heartbeater.